src/notebooks/notebookCommandListener.ts

```diff
@@ -175,6 +187,20 @@ export class NotebookCommandListener implements IDataScienceCommandListener {
         this.runAllCells();
     }
 
+    private async RestartKernelAndRunUpToSelectedCell(notebookUri: Uri | undefined) {
+        const uri = notebookUri ?? this.notebookEditorProvider.activeNotebookEditor?.notebook.uri;
+
+        await this.restartKernel(uri);
+        if (this.notebooks.activeNotebookEditor) {
+            this.commandManager
```

Comment:
I think we should capture this information before line 193 (i.e. before we restart), as its possible for focus to change while we're restarting (sure its unlikely, but who knows what `restartKernel` could do in the future).